src/rules.jl

```diff
@@ -25,26 +25,26 @@
 @define_diffrule Base.sec(x)                  = :(  sec($x) * tan($x)                  )
 @define_diffrule Base.csc(x)                  = :( -csc($x) * cot($x)                  )
 @define_diffrule Base.cot(x)                  = :( -(1 + cot($x)^2)                    )
-@define_diffrule Base.sind(x)                 = :(  (π / 180) * cosd($x)               )
-@define_diffrule Base.cosd(x)                 = :( -(π / 180) * sind($x)               )
-@define_diffrule Base.tand(x)                 = :(  (π / 180) * (1 + tand($x)^2)       )
-@define_diffrule Base.secd(x)                 = :(  (π / 180) * secd($x) * tand($x)    )
-@define_diffrule Base.cscd(x)                 = :( -(π / 180) * cscd($x) * cotd($x)    )
-@define_diffrule Base.cotd(x)                 = :( -(π / 180) * (1 + cotd($x)^2)       )
-@define_diffrule Base.sinpi(x)                = :(  π * cospi($x)                      )
-@define_diffrule Base.cospi(x)                = :( -π * sinpi($x)                      )
+@define_diffrule Base.sind(x)                 = :(  (oftype($x, π) / 180) * cosd($x)               )
```

Comment:
Should we be concerned what happens if `x` is a substantial expression? This might evaluate it twice, which could be avoided by something like
```
@gensym z
@define_diffrule Base.sind(x)  = :(  $z=$x;  (oftype($z, π) / 180) * cosd($z)   )
```
I see that a few existing rules do already violate this, e.g. with `exp(-$x * $x)`. 

It's possible that `@define_diffrule` could be extended to take care of this. It's also possible that every downstream use calls CSE which will clean this up anyway.

Reply:
I wasn't aware that `x` could potentially be an non-evaluated expression; if so, that's for sure an issue, but maybe a separate one from this PR?